_code-samples/modular-tutorials/create-offer.js

```diff
@@ -0,0 +1,125 @@
+/***********************************
+ *********** Create Offer **********
+ **********************************/
+
+ async function createOffer() {
+  let net = getNet()
+  let results = 'Connecting to ' + net + '....\n'
+  const client = new xrpl.Client(net)
+  await client.connect()
+      
+  results  += "Connected. Getting wallets.\n"
+  resultField.value = results
+  const wallet = xrpl.Wallet.fromSeed(accountSeedField.value)
+  results += accountNameField.value + " account address: " + wallet.address + "\n"
+  resultField.value = results
+
+
+  if (getCurrencyField.value == 'XRP') {
+    takerGets = getAmountField.value
+  } else {
+    takerGetsString = '{"currency": "' + getCurrencyField.value +'",\n' +
+        '"issuer": "' + getIssuerField.value + '",\n' +
+        '"value": "' + getAmountField.value + '"}'
+    takerGets = JSON.parse(takerGetsString)
+  }
+
+  if (payCurrencyField.value == 'XRP') {
+    takerPays = payAmountField.value
+  } else {
+    takerPaysString = '{"currency": "' + payCurrencyField.value + '",\n' +
+      '"issuer": "' + payIssuerField.value + '",\n' +
+      '"value": "' + payAmountField.value + '"}'
+    takerPays = JSON.parse(takerPaysString)
+  }
+ 
+   // -------------------------------------------------------- Prepare transaction
+  const prepared = await client.autofill({
+    "TransactionType": "OfferCreate",
+    "Account": wallet.address,
+    "TakerGets": takerGets,
+    "TakerPays": takerPays
+  })   
+  // ------------------------------------------------- Sign prepared instructions
+  const signed = wallet.sign(prepared)
+ results += "\nSubmitting transaction...."
+  // -------------------------------------------------------- Submit signed blob
+  const tx = await client.submitAndWait(signed.tx_blob)
+  results  += "\nBalance changes: " + 
+  JSON.stringify(xrpl.getBalanceChanges(tx.result.meta), null, 2)
+  resultField.value = results
+
+  xrpBalanceField.value =  (await client.getXrpBalance(wallet.address))
+  getOffers()                 
+  client.disconnect()    
+} // End of createOffer()
+
+/***********************************
+ ************ Get Offers ***********
+ **********************************/
+
+async function getOffers() {
+  let net = getNet()
+  let results = 'Connecting to ' + net + '....\n'
+  const client = new xrpl.Client(net)
+  await client.connect()
+  results  += "Connected.\n"
+  const wallet = xrpl.Wallet.fromSeed(accountSeedField.value)
+  results += accountNameField.value + " account: " + wallet.address
+// -------------------------------------------------------- Prepare request
+
+  results += '\n\n*** Offers ***\n'
+  let offers
+  try {
+    const offers = await client.request({
+      method: "account_offers",
+      account: wallet.address,
+      ledger_index: "validated"
+    })
+    results += JSON.stringify(offers,null,2)
+  } catch (err) {
+      results += err
```

Comment:
```suggestion
  } catch (err) {
      resultField.value += `${err.message}`
```

This suggestion applies to all functions called by the test harness. Current behavior will just kind of lock up and do nothing if the user adds in a bad field value or is missing something. While it can be a pain to code in proper messaging for all error types, the basic returned error is at least something that gives the user feedback they did a step wrong. I think all functions should include the try, catch block and spit out the error message if there is one.

Reply:
I made these changes.